arch/risc-v/src/mpfs/Make.defs

```diff
@@ -34,6 +34,14 @@ CMN_CSRCS += riscv_mdelay.c riscv_udelay.c riscv_copyfullstate.c
 CMN_CSRCS += riscv_idle.c riscv_tcbinfo.c riscv_getnewintctx.c
 CMN_CSRCS += riscv_cpuindex.c
 
+# If the NuttX kernel runs in S-mode
+
+ifeq ($(CONFIG_ARCH_USE_S_MODE),y)
+CMN_ASRCS += riscv_exception_macros.S riscv_machine_trap.S riscv_context.S
```

Comment:
The macros file is needed I think, but machine trap not. I will remove that.

Reply:
Maybe so, I'll remove it

Reply:
To very your change, it's better to add a new defconfig to enable S-mode in this patch too.

Reply:
Unfortunately this will end in a compile error in the riscv_sbi functions I think. The defconfig is ready but it also includes CONFIG_BUILD_KERNEL=y which is done, but I have not yet included that for review (it is a LARGE patch too!). So, the defconfig will come later.

Reply:
If it's compile/assembler error, I don't believe that adding riscv_exception_macros.S to CMN_ASRCS can fix your problem. Please recheck your change again.